test/test-v2-to-v3.js

```diff
@@ -0,0 +1,85 @@
+import fs from 'fs';
+import path from 'path';
+import { compareFormSchemas } from '../src/fulcrum-schema';
+import chai from 'chai';
+
+chai.should();
+
+let v1Form = null;
+let v2Form = null;
+let v3Form = null;
+
+const fixture = (file) => fs.readFileSync(path.join('test', 'fixtures', file)).toString();
+
+beforeEach(() => {
+  v1Form = JSON.parse(fs.readFileSync(path.join('test', 'fixtures', 'form-v1.json'))).form;
+  v2Form = JSON.parse(fs.readFileSync(path.join('test', 'fixtures', 'form-v2.json'))).form;
+  v3Form = JSON.parse(fs.readFileSync(path.join('test', 'fixtures', 'form-v3.json'))).form;
+  v1Form.row_id = 67777;
+  v2Form.row_id = 67777;
+  v3Form.row_id = 67777;
+});
+
+const dumpScript = function(scripts) {
+  console.log('----------------------------------');
+  console.log(toSQL(scripts));
+  console.log('----------------------------------');
+};
+
+const toSQL = (scripts) => scripts.join('\n').trim();
+
+const OPTIONS = {
+  dialect: 'postgres',
+  tableSchema: 'organization_1',
+  includeMetadata: true
+};
+
+describe('v2 to v3 form schema generator', () => {
+  it('builds a schema diff for a new v2 form', () => {
+    const sql = compareFormSchemas(null, v2Form, OPTIONS);
+
+    const expected = fixture('postgres-create-form.sql');
+
+    toSQL(sql).should.eql(expected);
+  });
+
+  it('builds a schema diff for a new v3 form', () => {
+    const sql = compareFormSchemas(null, v3Form, OPTIONS);
+
+    const expected = fixture('v3-create-form.sql');
+
+    toSQL(sql).should.eql(expected);
+  });
+
+  it('builds a schema diff for a v1 to v2 form', () => {
```

Comment:
What about v1 to v3?

Reply:
In theory, that can't happen. Fulcrum will set all Forms to have a schema_version of "v2", so the only way to use "v1" is to pass it through options (which will cause both schemas to use the same version). I can add a test, but it's not going to have a lot of value.